src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -441,6 +451,202 @@ try
+// - fontSize: the number to convert into an axis value
+// - format: the IDWriteTextFormat3 to get the defined axes from
+// Return value:
+// - The fully formed axes vector
+#pragma warning(suppress : 26429) // the analyzer doesn't detect that our FAIL_FAST_IF_NULL macro \
+    // checks format for nullness
+std::vector<DWRITE_FONT_AXIS_VALUE> DxFontRenderData::GetAxisVector(const DWRITE_FONT_WEIGHT fontWeight,
+                                                                    const DWRITE_FONT_STRETCH fontStretch,
+                                                                    const DWRITE_FONT_STYLE fontStyle,
+                                                                    const float fontSize,
+                                                                    IDWriteTextFormat3* format)
+{
+    FAIL_FAST_IF_NULL(format);
+
+    const auto axesCount = format->GetFontAxisValueCount();
+    std::vector<DWRITE_FONT_AXIS_VALUE> axesVector;
+    axesVector.resize(axesCount);
+    format->GetFontAxisValues(axesVector.data(), axesCount);
+
+    auto axisTagPresence = AxisTagPresence::None;
+    for (const auto& fontAxisValue : axesVector)
+    {
+        switch (fontAxisValue.axisTag)
+        {
+        case DWRITE_FONT_AXIS_TAG_WEIGHT:
+            WI_SetFlag(axisTagPresence, AxisTagPresence::Weight);
+            break;
+        case DWRITE_FONT_AXIS_TAG_WIDTH:
+            WI_SetFlag(axisTagPresence, AxisTagPresence::Width);
+            break;
+        case DWRITE_FONT_AXIS_TAG_ITALIC:
+            WI_SetFlag(axisTagPresence, AxisTagPresence::Italic);
+            break;
+        case DWRITE_FONT_AXIS_TAG_SLANT:
+            WI_SetFlag(axisTagPresence, AxisTagPresence::Slant);
+            break;
+        case DWRITE_FONT_AXIS_TAG_OPTICAL_SIZE:
+            WI_SetFlag(axisTagPresence, AxisTagPresence::OpticalSize);
+            break;
+        }
+    }
+
+    if (WI_IsFlagClear(axisTagPresence, AxisTagPresence::Weight))
+    {
+        axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{ DWRITE_FONT_AXIS_TAG_WEIGHT, gsl::narrow<float>(fontWeight) });
+    }
+    if (WI_IsFlagClear(axisTagPresence, AxisTagPresence::Width))
+    {
+        axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{ DWRITE_FONT_AXIS_TAG_WIDTH, FontStretchToWidthAxisValue(fontStretch) });
+    }
+    if (WI_IsFlagClear(axisTagPresence, AxisTagPresence::Italic))
+    {
+        axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{ DWRITE_FONT_AXIS_TAG_ITALIC, (fontStyle == DWRITE_FONT_STYLE_ITALIC ? 1.0f : 0.0f) });
+    }
+    if (WI_IsFlagClear(axisTagPresence, AxisTagPresence::Slant))
+    {
+        axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{ DWRITE_FONT_AXIS_TAG_SLANT, FontStyleToSlantFixedAxisValue(fontStyle) });
+    }
+    if (WI_IsFlagClear(axisTagPresence, AxisTagPresence::OpticalSize))
+    {
+        axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{ DWRITE_FONT_AXIS_TAG_OPTICAL_SIZE, DIPsToPoints(fontSize) });

```

Comment:
huh! so the default optical size is the font size in points?

Reply:
okay so in this case... i went and looked at Edge's code, which is similar to this (with axis tag presence).

They only use the font size in DIPs if the caller has _requested_ optical sizing. Otherwise, they ignore this tag completely (unless the user specifies it of course).

What does DirectWrite do by default? I wouldn't guess that it uses optical sizing by default...

Reply:
Ah, it looks like dwrite defaults to `DWRITE_AUTOMATIC_FONT_AXES_NONE` (which does NOT include OPSZ).

This one I would consider leaving off unless users specify :smile:

Reply:
Yes, I got this from noting what DWrite does (using reSearch) -

`float pointSize = DIPsToPoints(fontSize);
 completedFontAxisValues[newFontAxisValueCount++] = { DWRITE_FONT_AXIS_TAG_OPTICAL_SIZE, pointSize };`

And 

`// Converts from "device-independent pixel" units (1/96 inch) to points (1/72 inch).
inline float DIPsToPoints(float n)
{
    return n * (72.0f / 96.0f);
}`

Reply:
So, our renderer uses device-dependent pixels (or just pixels I guess we call them) or points in DX. You'll need to confirm that... but if we ditch opsz auto generation we don't need to worry about it :)

Reply:
We call this with `IDWriteTextFormat`'s `GetFontSize` which returns the font size in DIPs according to the [docs](https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritetextformat-getfontsize)

Reply:
Removed the opsz auto generation for now in any case!